packages/dataviews/src/components/dataviews-view-config/infinite-scroll-toggle.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import DataViewsContext from '../dataviews-context';
+
+export default function InfiniteScrollToggle() {
+	const context = useContext( DataViewsContext );
+	const { view, onChangeView } = context;
+	const infiniteScrollEnabled = view.layout?.infiniteScroll ?? false;
+
+	// Only render the toggle if an infinite scroll handler is available
+	if ( ! context.hasInfiniteScrollHandler ) {
+		return null;
+	}
+
+	return (
+		<ToggleGroupControl
+			__nextHasNoMarginBottom
+			__next40pxDefaultSize
+			isBlock
+			label={ __( 'Infinite scroll' ) }
+			value={ infiniteScrollEnabled ? 'enabled' : 'disabled' }
+			onChange={ ( value ) => {
+				const newValue = value === 'enabled';
+				onChangeView( {
```

Comment:
When switching it off, we also need to set the page we are in? Otherwise, it's like nothing happened:

https://github.com/user-attachments/assets/b287dba2-8226-4b15-92b6-c0d02fa83af8